sources/api/host-containers/src/main.rs

```diff
@@ -434,22 +434,47 @@ where
     Ok(())
 }
 
+fn is_restart_required(changed_settings: &String, container_name: &str) -> bool {
+    if changed_settings.is_empty() {
+        // it means that Bottlerocket is booting - all containers need to be started
+        return true;
+    }
+
+    let settings: Vec<&str> = changed_settings.split_whitespace().collect();
+    let setting_prefix = format!("host-containers.{}", container_name);
+
+    for setting in settings.iter() {
+        if setting.contains(&setting_prefix) {
+            info!("Found a changed setting for {} container", container_name);
+            return true;
+        }
+        if !setting.contains("host-containers") {
+            // if its some other setting, return true for all host-containers, example: network
+            return true;
+        }
+    }
+    return false;
+}
+
 async fn run() -> Result<()> {
     let args = parse_args(env::args());
+    let changed_settings = env::var("changed_settings").unwrap_or("".to_string());
```

Comment:
Nit: It will be good to document that this environment variable is passed by `thar-be-settings`